### DISCUSSION TITLE
Backend postgresql configuration when using skywalking-kubernetes

### DISCUSSION THREAD

**Discussion by ghost** [apache/skywalking#8607] [Q&A] [answered]
Hello,

I've been checking `skywalking-kubernetes` repository to deploy the application via Helm, but I don't see any template for the PostgreSQL backend configuration. I understand the storage backend configuration goes inside the `skywalking-oap-server` image, under the `config/application.yml` file, but I was wondering which would be the best approach.

With the current state I think you have two choices:

1. Manually download the repo, update the configuration and build the image.
2. Save all files under `config` folder inside a configMap for example and mount that configMap in your deployment, so you can use the image already posted in DockerHub and just need to tune the configuration files.

With the above choices I was wondering if its possible to enable a simpler way to override the configuration or if you have any other simpler way of doing it. The idea, since we're using Helm, is to have all in templates if possible, to touch as little as possible manually (at least regarding on the deployment part).

**Comment by wu-sheng** [comment, marked as answer]
Generally, all configs are exposed through system environments, I think they are open in helm already.

https://skywalking.apache.org/docs/main/v8.9.1/en/setup/backend/configuration-vocabulary/
All names are available here.

**Reply by ghost** [reply]
Ah, my bad, I saw that the connection properties where referenced in the `application.yaml` file and I thought you could only configure them inside the file itself, but the file tries to read an environment variable first.

Regarding the `skywalking-kubernetes` part, would it be possible to add this values to the chart itself as a new feature/enhancement? So the chart already applies the required env config (as it does for ElasticSearch for example) instead of having to apply the Helm template and then patching it to add those values.

**Reply by wu-sheng** [reply]
I would say that would be too scary to new users, feel overwhelming about these keys.
And these configs are changing in every OAP release, we should not release helm from time to time just for these.

**Reply by ghost** [reply]
Yeah, that's understandable since its a lot of configuration depending on the scenario/storage type. Thanks for the clarification.

**Reply by warrmr** [reply]
> I would say that would be too scary to new users, feel overwhelming about these keys. And these configs are changing in every OAP release, we should not release helm from time to time just for these.

It's already big and scary! :( I'm looking for a TL;DR on the K8s install of Skywalking with Postgresql or InfluxDB. Totally new to K8s and Skywalker and am trying to build a POC completely blind talk about trial by fire :)
